[Refactor] 1주차 회의 리팩토링 진행

windowM opened the pull request:
- 토큰 유효성 확인하는 로그(info lev)삭제 -> 토큰 관련 예외는 `parseClaim()` 에서 던진다.
- `userdetails` -> `getUsername()` 반환값 수정 ( email -> id)
-  `UrlPermissionChecker` 인증 필요없는 uri들 별도 클래스에 보관
  - 모든 http 메소드에 대해 정의해줘야 하는게 싫어서 Map 사용.

windowM committed:
[docs] application-local.yml 수정

- 토큰 만료시간 수정

windowM committed:
[refactor] 1주차 리팩토링 진행

<JwtProvider>
- 토큰 유효성 검증 관련
  - 예외는 parseCalim()에서 던진다
  - checkExpiredToken() -> isExpiredToken() 이름 및 반환 타입 변경 (예외는 service에서 던진다.)

<Member>
- getUsername() 반환값 email -> id로 수정 (oauth2)

<CustomUserDetails>
- 필요없는 메소드 삭제

<UrlPermissionChecker>
- 인증이 필요없는 경로들에 대해 필터에서 걸러지지 않게 하기 위한 permission url경로들이 저장된 별도 클래스 파일 정의

windowM committed:
[docs] api prefix 설정

Kong-TaeHyeon commented on src/main/java/com/example/fastboard/domain/member/service/AuthService.java:
```diff
@@ -46,7 +48,9 @@ public TokenResponse reissueAccessToken(RefreshTokenRequest request) {
 
         // 이전에 발급된 액세스 토큰이 만료가 되어야 새로운 액세스 토큰 발급
-        jwtProvider.checkExpiredToken(oldAccessToken);
+        if(!jwtProvider.isExpiredToken(oldAccessToken)){
```
재발급 요청에서 현재 AccessToken 이 만료가 되어야 하는 이유가 무엇인지 궁금합니다!
저는 따로 유효기간을 체크하지 않아서요!!

windowM replied:
엇 이전 pr 커멘트에 남겼던거 같은데, AccessToken이 만료되지 않았는데 재발급 해야하는 이유가 있나요?
저는 당연히 만료가 되야지 재발급을 해야하는거라 생각해서 체크를 진행했습니다..!

Kong-TaeHyeon replied:
기존 AccessToken 이 탈취당한 경우, 해당 토큰으로 계속해서 만료 시간이 도달하기 전까지 공격자가 서비스를 이용할 수 있지 않을까요?
* 그래서 저는 만료를 확인하는 것이 아닌, 기존 AccessToken 을 무효화 시키도록 해야 한다고 생각했었어요!

windowM replied:
음.. 확실히 탈취된 경우에 한해서 재발급을 받으려 할 때 문제가 있겠네요..
그런데 혹시 악의적으로 무한정으로 재발급 받으려고 하는 부분에서 생기는 문제에 대해서는 어떻게 막을 수 있을까요?

Kong-TaeHyeon commented on src/main/java/com/example/fastboard/domain/member/service/CustomUserDetailsService.java:
```diff
@@ -18,17 +16,6 @@ public class CustomUserDetailsService implements UserDetailsService {
-                .password(passwordEncoder.encode(member.getPassword()))
-                .roles(String.valueOf(member.getRole()))
-                .build();
+        return memberService.findActiveMemberByEmail(email);
```
email 로 기준하여 멤버를 찾고 계시는데 만약 소셜로그인이 함께 구현되는 상황이라면 어떻게 처리하실지 궁금합니다!
* user1@gmail.com => Google 을 사용한 소셜 로그인 이메일
* user1@gmail.com => 해당 서비스의 계정 user1@gmail.com

저라면, findActiveMemberByEmail 에 type 을 파라미터로 추가할 것 같은데 혹시 또 다른 생각이 있으실까요?

windowM replied:
우선 오어스가 적용되면 어떤식으로 유저엔티티에 데이터가 담기게 되고 어떤 메커니즘으로 굴러가는지(?) 잘 몰라서 생각하고 구현한건 아닙니다!

한 가지 궁금한 것은 오어스 이메일과 서비스 계정 아이디(이메일) 중복 여부가 가능한가요?
가능하다면, type을 추가하는 방법도 좋을것 같습니다만, 
오어스 추가됨에 따라 일어나는 부분은 제가 학습을 해야 할 것 같습니다!

Kong-TaeHyeon replied:
`오어스 이메일과 서비스 계정 아이디(이메일) 중복 여부가 가능한가요?`
이는 선택사항인 것 같아요!! 저희가 사용하는 서비스들을 보면 기존 서비스는 이메일이 아닌 아이디, 패스워드 형식이고 OAuth 는 이메일 형식으로 로그인하는 경우가 대부분인데, 저희는 자체 로그인 또한 이메일 형식으로 지원하다 보니, 나중에 OAuth 계정의 이메일을 저장할 때는 어떻게 처리해야 할 지에 대해서 Type 컬럼을 제안한 것이었습니다!

windowM replied:
아하!
흠 그런데 제 생각에 요즘은 이메일을 아이디로 사용하는 사이트들이 많지 않나요?
별도 사이트id없이 이메일로만 회원가입을 했던 사이트들이 많이 보였던 기억이 있는데 잘 모르겠네요...

type을 사용하는 방법이나, 기획을 할 때 사이트 계정은 별도 id를 받는 2가지 방법이 있나보네요
좋은 의견 감사합니다! 이 부분은 제가 OAuth를 구현해봐야 고려해 볼 문제가 될 거 같네요!

windowM committed:
[feat] Board엔티티 Category enum 상수 -> 문자열로 변환

windowM committed:
[feat] 게시글 생성 기능 추가

windowM committed:
[fix] 토큰 추출 메소드에서 토큰이 없다면 예외 던지기

windowM committed:
[test] 게시글 생성 단위 테스트

windowM committed:
[feat] 유효성 검사 추가

windowM committed:
[feat] 이미지 파일 저장될 경로 지정(local)

windowM committed:
[feat] 이미지 업로드 기능 구현

- 멤버 조회
- 파일 유효성 검사(null,용량)
- 원본 파일명 get(비어있을 경우 예외)
- 유니크 파일명 생성 (UUID)
- Path 지정
- 디렉토리 파일 없다면 생성
- 파일 저장
- db 저장
- img저장 경로 return

windowM committed:
[chore] 코드 정리

windowM committed:
[feat] 게시글 전체 조회

windowM committed:
[feat] 게시글 이미지 로직 변경

- 이미지 업로드 api 방식 유지
- 게시글 저장시 content내부에 <img>태그를 파싱 후 src="" 내부 경로만 추출
- 추출한 경로를 바탕으로 file이름을 찾고 image db에서 board를 update해줌

windowM committed:
Merge pull request #29 from fast-board/feature/7-boards-list

[Feature] /boards 게시글 전체 조회 & 게시판 이미지 처리 로직 수정

windowM committed:
Merge pull request #28 from fast-board/feature/4-images-upload

[Feature] /images 이미지 업로드 기능 구현

windowM committed:
Merge pull request #26 from fast-board/feature/5-board-create

[Feature] /boards 게시글 작성 기능 구현
